src/helpers/uuid_helper.coffee

```diff
@@ -1,35 +1,86 @@
 ###
-Tiny library to generate version 4 UUIDs
+Library to generate version 4 UUIDs
+
+Code taken from:
+  https://github.com/kelektiv/node-uuid/tree/v3.3.3
 
 References:
   http://www.ietf.org/rfc/rfc4122.txt
-  https://github.com/kelektiv/node-uuid
-  https://gist.github.com/jed/982883
 ###
 define ->
-  cryptoObj = window.crypto or window.msCrypto # for IE 11
+  UUID =
+    # _cryptoObj, _rng(), _whatwgRNG() and _mathRNG() were taken from:
+    # https://github.com/kelektiv/node-uuid/blob/v3.3.3/lib/rng-browser.js
+    _cryptoObj: window.crypto or window.msCrypto # for IE 11
 
-  if cryptoObj
-    random = ->
-      # Divide a random UInt32 by the maximum value (2^32 -1) to get a result between 0 and 1
-      # https://developer.mozilla.org/en-US/docs/Web/API/Crypto/getRandomValues
-      cryptoObj.getRandomValues(new Uint32Array(1))[0] / 4294967295
-  else
-    random = Math.random
+    # Unique ID creation requires a high quality random # generator.  In the
+    # browser this is a little complicated due to unknown quality of Math.random()
+    # and inconsistent support for the `crypto` API.  We do the best we can via
+    # feature-detection
+    _rng: () -> if UUID._cryptoObj then UUID._whatwgRNG() else UUID._mathRNG()
 
-  # algorithm from https://gist.github.com/jed/982883
-  UUID =
-    generate: (placeholder) ->
-      if placeholder
-        # if the placeholder was passed,
-        # return a random number from 0 to 15
-        # unless placeholder is 8,
-        # in which case a random number from 8 to 11
-        # in hexadecimal
-        (placeholder ^ random() * 16 >> placeholder/4).toString(16)
-      else
-        # or otherwise a template string replacing
-        # zeroes, ones, and eights with random hex digits
-        '10000000-1000-4000-8000-100000000000'.replace(/[018]/g, UUID.generate)
+    # WHATWG crypto RNG - http://wiki.whatwg.org/wiki/Crypto
+    _whatwgRNG: () ->
+      rnds8 = new Uint8Array(16);
+
+      UUID._cryptoObj.getRandomValues(rnds8)
+
+      rnds8
+
+    # Math.random()-based (RNG)
+
+    # If all else fails, use Math.random().  It's fast, but is of unspecified
+    # quality.
+    _mathRNG: () ->
+      rnds = new Array(16);
+
+      for i in [0...16]
+        r = Math.random() * 0x100000000 if (i & 0x03) == 0
+        rnds[i] = r >>> (i & 0x03) << 3 & 0xff
+
+      rnds
+
+    # _bytesToUuid() and _byteToHex were taken from:
+    # https://github.com/kelektiv/node-uuid/blob/v3.3.3/lib/bytesToUuid.js
+    #
+    # Convert array of 16 byte values to UUID string format of the form:
+    # XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX
+    _byteToHex: (byteToHex = []; byteToHex[i] = (i + 0x100).toString(16).substr(1) for i in [0...256])
```

Comment:
`byteToHex` and `i` are leaked (`payload.js:3587`)

Personally, I think this is fine

![this-is-fine 0](https://user-images.githubusercontent.com/6825357/64694688-f0211080-d4a2-11e9-9fb6-5e0edefe5bc9.jpg)

Reply:
Nice catch!
I used a self invoking function.